kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/history/presenter/CouponDialogPresenter.java

```diff
@@ -3,26 +3,37 @@
 import android.support.annotation.NonNull;
 import com.kin.ecosystem.R;
 import com.kin.ecosystem.base.BaseDialogPresenter;
-import com.kin.ecosystem.base.IBottomDialogPresenter;
+import com.kin.ecosystem.bi.EventLogger;
+import com.kin.ecosystem.bi.events.RedeemUrlTapped;
+import com.kin.ecosystem.bi.events.SpendRedeemButtonTapped;
+import com.kin.ecosystem.bi.events.SpendRedeemPageViewed;
+import com.kin.ecosystem.bi.events.SpendRedeemPageViewed.RedeemTrigger;
 import com.kin.ecosystem.data.model.Coupon;
 import com.kin.ecosystem.data.model.Coupon.CouponInfo;
 import com.kin.ecosystem.history.view.ICouponDialog;
+import com.kin.ecosystem.network.model.Order;
 
-public class CouponDialogPresenter extends BaseDialogPresenter<ICouponDialog> implements
-    IBottomDialogPresenter<ICouponDialog> {
+public class CouponDialogPresenter extends BaseDialogPresenter<ICouponDialog> implements ICouponDialogPresenter {
 
+	private final EventLogger eventLogger;
     private final Coupon coupon;
+    private final Order order;
+    private final RedeemTrigger redeemTrigger;
     private static final String HTTP_URL_PATTERN= "http://";
     private static final String HTTPS_URL_PATTERN= "https://";
 
-    public CouponDialogPresenter(@NonNull final Coupon coupon) {
+    CouponDialogPresenter(@NonNull final Coupon coupon, @NonNull Order order, RedeemTrigger redeemTrigger, @NonNull EventLogger eventLogger) {
+        this.eventLogger = eventLogger;
         this.coupon = coupon;
+        this.order = order;
+        this.redeemTrigger = redeemTrigger;
     }
 
     @Override
     public void onAttach(ICouponDialog view) {
         super.onAttach(view);
-        loadInfo();
+		loadInfo();
+		eventLogger.send(SpendRedeemPageViewed.create(redeemTrigger, (double) order.getAmount(), order.getOfferId(), order.getOrderId()));
```

Comment:
check order!= null?

Reply:
order marked on the constructor as nonnull